build with gcc-15 fails to build with 1.7.3 and master.

```
grammar.y: In function 'capture_parse_pos':
grammar.y:60:33: error: too many arguments to function 'libconfig_yyget_lineno'
   60 |   setting->line = (unsigned int)libconfig_yyget_lineno(scanner);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
grammar.y:44:12: note: declared here
   44 | extern int libconfig_yyget_lineno();
      |            ^~~~~~~~~~~~~~~~~~~~~~
grammar.y: In function 'libconfig_yyerror':
grammar.y:71:29: error: too many arguments to function 'libconfig_yyget_lineno'
   71 |   ctx->config->error_line = libconfig_yyget_lineno(scanner);
      |                             ^~~~~~~~~~~~~~~~~~~~~~
grammar.y:44:12: note: declared here
   44 | extern int libconfig_yyget_lineno();
      |            ^~~~~~~~~~~~~~~~~~~~~~
grammar.c: At top level:
grammar.c:238:5: error: conflicting types for 'libconfig_yylex'; have 'int(YYSTYPE *, void *)'
  238 | int libconfig_yylex (YYSTYPE *yylvalp, void *scanner);
      |     ^~~~~~~~~~~~~~~
grammar.y:43:12: note: previous declaration of 'libconfig_yylex' with type 'int(void)'
   43 | extern int libconfig_yylex();
      |            ^~~~~~~~~~~~~~~
make[3]: *** [Makefile:616: libconfig___la-grammar.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
grammar.y: In function 'capture_parse_pos':
grammar.y:60:33: error: too many arguments to function 'libconfig_yyget_lineno'
   60 |   setting->line = (unsigned int)libconfig_yyget_lineno(scanner);
      |                                 ^~~~~~~~~~~~~~~~~~~~~~
grammar.y:44:12: note: declared here
   44 | extern int libconfig_yyget_lineno();
      |            ^~~~~~~~~~~~~~~~~~~~~~
grammar.y: In function 'libconfig_yyerror':
grammar.y:71:29: error: too many arguments to function 'libconfig_yyget_lineno'
   71 |   ctx->config->error_line = libconfig_yyget_lineno(scanner);
      |                             ^~~~~~~~~~~~~~~~~~~~~~
grammar.y:44:12: note: declared here
   44 | extern int libconfig_yyget_lineno();
      |            ^~~~~~~~~~~~~~~~~~~~~~
grammar.c: At top level:
grammar.c:238:5: error: conflicting types for 'libconfig_yylex'; have 'int(YYSTYPE *, void *)'
  238 | int libconfig_yylex (YYSTYPE *yylvalp, void *scanner);
      |     ^~~~~~~~~~~~~~~
grammar.y:43:12: note: previous declaration of 'libconfig_yylex' with type 'int(void)'
   43 | extern int libconfig_yylex();
      |            ^~~~~~~~~~~~~~~
make[3]: *** [Makefile:672: libconfig_la-grammar.lo] Error 1
```

--- a/lib/grammar.y
+++ b/lib/grammar.y
@@ -40,8 +40,7 @@
 #include "wincompat.h"
 
 /* These declarations are provided to suppress compiler warnings. */
-extern int libconfig_yylex();
-extern int libconfig_yyget_lineno();
+extern int libconfig_yyget_lineno(void *);
 
 #define YYMALLOC libconfig_malloc
 
@@ -81,6 +80,11 @@ void libconfig_yyerror(void *scanner, struct parse_context *ctx,
   double fval;
   char *sval;
 }
+
+%{
+/* These declarations are provided to suppress compiler warnings. */
+extern int libconfig_yylex(YYSTYPE *, void *);
+%}
 
 %token <ival> TOK_BOOLEAN TOK_INTEGER TOK_HEX
 %token <llval> TOK_INTEGER64 TOK_HEX64

